src/common/common.ts

```diff
@@ -396,6 +404,356 @@ export async function getVerifiableCredentialApiConfiguration(
+
+  const contentType = response.headers.get("Content-Type");
+  if (!contentType) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: Response does not have a Content-Type header; expected application/ld+json`,
+    );
+  }
+
+  const parsedContentType = contentTypeParser.parse(contentType);
+  const [mediaType, subtypesString] = parsedContentType.type.split("/");
+  const subtypes = subtypesString.split("+");
+
+  if (mediaType !== "application" || !subtypes.includes("json")) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: Response has an unsupported Content-Type [${contentType}]; expected application/ld+json`,
+    );
+  }
+
+  return response;
+}
+
+async function responseToVcStore(
+  response: Response,
+  vcUrl: UrlString,
+  options?: Partial<{
+    fetch: typeof fetch;
+  }>,
+): Promise<Store> {
+  try {
+    return await jsonLdResponseToStore(validateVcResponse(response, vcUrl), {
+      fetch: options?.fetch,
+      baseIRI: vcUrl,
+    });
+  } catch (e) {
+    throw new Error(
+      `Parsing the Verifiable Credential [${vcUrl}] as JSON-LD failed: ${e}`,
+    );
+  }
+}
+
+const VC = "https://www.w3.org/2018/credentials#";
+const RDF = "http://www.w3.org/1999/02/22-rdf-syntax-ns#";
+const XSD = "http://www.w3.org/2001/XMLSchema#";
+const SEC = "https://w3id.org/security#";
+const CRED = "https://www.w3.org/2018/credentials#";
+const RDF_TYPE = `${RDF}type`;
+const VERIFIABLE_CREDENTIAL = `${VC}VerifiableCredential`;
+const DATE_TIME = `${XSD}dateTime`;
+const CREDENTIAL_SUBJECT = `${CRED}credentialSubject`;
+const ISSUER = `${CRED}issuer`;
+const ISSUANCE_DATE = `${CRED}issuanceDate`;
+const PROOF = `${SEC}proof`;
+const PROOF_PURPOSE = `${SEC}proofPurpose`;
+const VERIFICATION_METHOD = `${SEC}verificationMethod`;
+const PROOF_VALUE = `${SEC}proofValue`;
+
+/**
+ * @hidden
+ */
+export async function getVerifiableCredentialFromStore(
+  vcStore: Store,
+  vcUrl: UrlString,
+): Promise<VerifiableCredential & DatasetCore> {
+  const vcContext = await getVcContext();
+
+  const vcs = vcStore.getSubjects(
+    RDF_TYPE,
+    VERIFIABLE_CREDENTIAL,
+    DF.defaultGraph(),
+  );
+  if (vcs.length !== 1) {
+    throw new Error(
+      `Expected exactly one Verifiable Credential in [${vcUrl}], received: ${vcs.length}`,
+    );
+  }
+
+  const [vc] = vcs;
+  if (vc.termType !== "NamedNode") {
+    throw new Error(
+      `Expected the Verifiable Credential in [${vcUrl}] to be a Named Node, received: ${vc.termType}`,
+    );
+  }
+
+  const type: string[] = [];
+  for (const t of vcStore.getObjects(vc, RDF_TYPE, DF.defaultGraph())) {
+    if (t.termType !== "NamedNode") {
+      throw new Error(
+        `Expected all VC types to be Named Nodes but received [${t.value}] of termType [${t.termType}]`,
+      );
+    }
+
+    const compact = vcContext.compactIri(t.value, true);
+
+    if (/^[a-z]+$/i.test(compact)) type.push(compact);
+  }
+
+  function getSingleObject(
```

Comment:
Do these functions need to be inlined? `getVerifiableCredentialFromStore` is already pretty massive, moving these out would make it a bit more readable, wouldn't it?